### PR TITLE
Add missing dispatch group leave in case of download artifacts download fail

### PR DESCRIPTION
**Pull request by krzyzanowskim**
Add missing dispatch group leave in case of download artifacts download fail

Leave dispatch group in all cases - unstuck download queue in case of failure.

### CODE REVIEW

File: Sources/Workspace/Workspace.swift [hunk 1]
```diff
@@ -1619,24 +1619,27 @@ extension Workspace {
             group.enter()
 
             guard case .remote(let url, let checksum, _) = artifact.source, let destination = path(for: artifact) else {
+                group.leave()
```
Comment (tomerd) [inline Sources/Workspace/Workspace.swift:1622]: good catch. how about a `defer { group.leave() }` next to `group.enter()` then a second enter/leave pair for downloader.downloadFile?

Comment (krzyzanowskim) [reply Sources/Workspace/Workspace.swift]: Either way works for me 👍

File: Sources/Workspace/Workspace.swift [hunk 2]
```diff
@@ -1656,17 +1660,16 @@ extension Workspace {
                         guard archiveChecksum == checksum else {
                             tempDiagnostics.emit(.artifactInvalidChecksum(targetName: artifact.targetName, expectedChecksum: checksum, actualChecksum: archiveChecksum))
                             tempDiagnostics.wrap { try self.fileSystem.removeFileTree(archivePath) }
-                            group.leave()
                             return
                         }
 
+                        group.enter()
                         self.archiver.extract(from: archivePath, to: parentDirectory, completion: { extractResult in
```
Comment (tomerd) [inline Sources/Workspace/Workspace.swift:1667]: one last thing: add a `defer { group.leave() }` here instead of L1679. otherwise 👍

**Review by tomerd — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Review by abertelrud — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci please smoke test macOS Platform

**Comment by tomerd** [comment]
@swift-ci please smoke test Linux

**Comment by tomerd** [comment]
@krzyzanowskim could you please resolve the conflict so we can merge this

**Comment by krzyzanowskim** [comment]
@tomerd done. thanks!

**Comment by tomerd** [comment]
@swift-ci please smoke test

File: Sources/Workspace/Workspace.swift (hunk 2 above)
Comment (tomerd) [reply Sources/Workspace/Workspace.swift:1673]: sorry @krzyzanowskim still need to address this little bit

Comment (krzyzanowskim) [reply Sources/Workspace/Workspace.swift:1673]: done

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Pull request merged**